test/Microsoft.AspNetCore.Server.KestrelTests/MemoryPoolIteratorTests.cs

```diff
@@ -1214,6 +1214,71 @@ public void TestGetAsciiStringEscaped(string input, string expected, int maxChar
             }
         }
 
+        [Fact]
+        public unsafe void TestCopyFromAscii()
```

Comment:
I'd like to see some boundary testing here, since you're copying 8 bytes at a time and then whatever is left.